server/src/test/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequestTests.java

```diff
@@ -118,65 +112,4 @@ public void testMappingKeyedByType() throws IOException {
             assertEquals(request1.mappings(), request2.mappings());
         }
     }
-
-    @Override
-    protected PutIndexTemplateRequest createTestInstance() {
```

Comment:
It looks like the server side PutIndexTemplateRequest still supports internal wire serialization, which I think we should still test using random instances. Maybe changing this test to one of the AbstractWire*TestCase (tbh I'm getting confused which one would be appropriate) would be good, as long as we are able to detect faulty writeTo/readFrom implementation if we e.g. add new parameters.

Reply:
I had initially tried this out, but ended up uncovering some small bugs in `PutIndexTemplateRequest`. So I planned to defer the change to a follow-up PR -- is that okay with you? Currently the test just extends `AbstractXContentTestCase`, so it doesn't test wire serialization and I don't think we are regressing test coverage in that regard.

Reply:
No problem, I assumed AbstractXContentTestCase would also include wire serialization tests. I was wrong, but thats part of why I dislike our growing AbstractWire/XContent/*/TestCase family of base tests, because after a while its hard to remember what exactly they are doing. If you could open an issue that we should test serialization here and maybe add your findings about the "small bugs" that you found, that would be great.

Reply:
👍